src/libraries/Common/src/Interop/Windows/Dnsapi/Interop.DnsRecordListFree.cs

```diff
@@ -0,0 +1,16 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Dnsapi
+    {
+        [LibraryImport(Libraries.Dnsapi)]
+        internal static partial void DnsRecordListFree(
+            IntPtr dnsResultList,
+            [MarshalAs(UnmanagedType.Bool)] bool dnsFreeType);
```

Comment:
```suggestion
        [LibraryImport(Libraries.Dnsapi)]
        internal static partial void DnsRecordListFree(
            IntPtr dnsResultList,
            [MarshalAs(UnmanagedType.Bool)] bool dnsFreeType);
```

The above seems to be a bit odd. The `WinDNS.h` header defines this as follows:

```c
#if(_WIN32_WINNT >= 0x0501)
#define DnsRecordListFree(p,t)  DnsFree(p,DnsFreeRecordList)
#else
VOID
WINAPI
DnsRecordListFree(
    _Inout_opt_ PDNS_RECORD     pRecordList,
    _In_        DNS_FREE_TYPE   FreeType
    );
#endif /* _WIN32_WINNT >= 0x0501 */
```

The signature here is off and I _think_ this should just point to `DnsFree()` as we are long past Windows 2000.

Feel free to make this method `private` and keep the `DNS_FREE_TYPE` argument as an implementation detail as an `int`. You can then keep this signature as is, but implement it with `DnsFree()`.

Reply:
I was about to ask whether it would be preferable to just replace it wholesale (e.g. import `DnsFree`, define `enum DNS_FREE_TYPE : uint`, update call sites, etc....

But then the only two places `DnsRecordListFree` and `DnsQuery` are used are in `S.DS.AD.Locator.DnsQueryWrapper` which appears to be completely unused - should I just delete that private method, and purge the `Dnsapi` interop code altogether? It doesn't seem to be used in any other libraries...

```
Chris@DESKTOP-C8V4VRD MINGW64 ~/src/dotnet/runtime (system-directoryservices-interop-refactor)
$ grep -r --include '*.cs' DnsQueryWrapper src/libraries
src/libraries/System.DirectoryServices/src/System/DirectoryServices/ActiveDirectory/Locator.cs:        private static Hashtable DnsQueryWrapper(string domainName, string siteName, long dcFlags)

Chris@DESKTOP-C8V4VRD MINGW64 ~/src/dotnet/runtime (system-directoryservices-interop-refactor)
$ grep -r --include '*.cs' DnsQuery src/libraries
src/libraries/Common/src/Interop/Windows/Dnsapi/Interop.DnsQuery.cs:        [LibraryImport(Libraries.Dnsapi, EntryPoint = "DnsQuery_W", StringMarshalling = StringMarshalling.Utf16)]
src/libraries/Common/src/Interop/Windows/Dnsapi/Interop.DnsQuery.cs:        internal static partial int DnsQuery(
src/libraries/System.DirectoryServices/src/System/DirectoryServices/ActiveDirectory/Locator.cs:        private static Hashtable DnsQueryWrapper(string domainName, string siteName, long dcFlags)
src/libraries/System.DirectoryServices/src/System/DirectoryServices/ActiveDirectory/Locator.cs:                options |= NativeMethods.DnsQueryBypassCache;
src/libraries/System.DirectoryServices/src/System/DirectoryServices/ActiveDirectory/Locator.cs:            // Call DnsQuery
src/libraries/System.DirectoryServices/src/System/DirectoryServices/ActiveDirectory/Locator.cs:            result = Interop.Dnsapi.DnsQuery(recordName, NativeMethods.DnsSrvData, options, IntPtr.Zero, out dnsResults, IntPtr.Zero);
src/libraries/System.DirectoryServices/src/System/DirectoryServices/ActiveDirectory/NativeMethods.cs:        internal const int DnsQueryBypassCache = 8;

Chris@DESKTOP-C8V4VRD MINGW64 ~/src/dotnet/runtime (system-directoryservices-interop-refactor)
$ grep -r --include '*.cs' DnsRecordListFree src/libraries
src/libraries/Common/src/Interop/Windows/Dnsapi/Interop.DnsRecordListFree.cs:        internal static partial void DnsRecordListFree(
src/libraries/System.DirectoryServices/src/System/DirectoryServices/ActiveDirectory/Locator.cs:                        Interop.Dnsapi.DnsRecordListFree(dnsResults, true);
```

Reply:
> should I just delete that private method, and purge the Dnsapi interop code altogether?

+1 Yes! Thank you for looking at that.

Reply:
Nice, done in 8808eed2df5301b04268fdf2cdde9a47fb9b5f74

Reply:
It looks like you can also delete `PartialDnsRecord` and `DnsRecord` in the `NativeMethods.cs` file, can you confirm that?

Reply:
Yes indeed, along with some Dnsapi-related consts in there too - done in b5c22d673c003682851be052ffa80cfdbfd5bc9e